Assets/MRTK/SDK/Features/UX/Scripts/Buttons/ButtonConfigHelper.cs

```diff
@@ -414,7 +414,192 @@ public void ForceRefresh()
 
         private void OnEnable()
         {
+#if UNITY_EDITOR
+            if (EditorCheckForCustomIcon())
+            {   // If we're using a custom icon, preserve it so it doesn't vanish
+                EditorPreserveCustomIcon();
+            }
+#else
+            // Set these to null to avoid build errors.
+            defaultIconSet = null;
+            defaultButtonQuadMaterial = null;
+#endif
+
+            ForceRefresh();
+        }
+
+#if UNITY_EDITOR
+
+        private static readonly string generatedIconSetName = "CustomIconSet";
+        private static readonly string customIconSetsFolderName = "CustomIconSets";
+        private static readonly string customIconSetCreatedMessage = "A new icon set has been created to hold your button's custom icons. This icon set will be used by your button's ButtonConfigHelper component. It has been saved to:\n\n{0}";
+        
+        /// <summary>
+        /// Returns true if the button is using a custom icon material.
+        /// </summary>
+        public bool EditorCheckForCustomIcon()
+        {
+            if (iconSet == null || iconQuadRenderer == null || iconStyle != ButtonIconStyle.Quad)
+            {   // Nothing to do here.
+                return false;
+            }
+
+            if (iconSet != defaultIconSet)
+            {   // This button is using a custom icon set, so we can't assume material differences mean it needs an upgrade.
+                return false;
+            }
+
+            if (iconQuadRenderer.sharedMaterial == defaultButtonQuadMaterial)
+            {   // This button is using the default material, so it's not a customized button.
+                return false;
+            }
+
+            string assetPath = AssetDatabase.GetAssetPath(iconQuadRenderer.sharedMaterial);
+            if (string.IsNullOrEmpty(assetPath))
+            {   // If the asset path is null, this material instance exists only in memory.
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Upgrades a button using a custom icon material.
+        /// </summary>
+        public void EditorUpgradeCustomIcon()
+        {
+            if (iconSet == null || iconQuadRenderer == null || iconStyle != ButtonIconStyle.Quad)
```

Comment:
Why is this line doing a subset of the checks that are done in EditorCheckForCustomIcon? i.e. why aren't the other ones here as well (or rather, why not just call EditorCheckForCustomIcon if we're gonna do a duplication of some of the condition checks?)

Otherwise you run the risk of introducing bugs where you update one of the if clauses but not the other.